README.md

```diff
@@ -7,620 +7,799 @@
+values will be whatever we actually entered into our URL.
+
+If we click on `George Orwell's` card for example, our `params` object should
+look like this:
+
+```JavaScript
+{
+  id: "1"
 }
+```
 
-export default Login;
+We can now use the data contained in our params object to access the specific
+piece of data we want to display!
+
+```JavaScript
+const user = users.find(user => user.id === parseInt(params.id))
 ```
 
-```jsx
-// src/components/NavBar.js
-import React from "react";
-import { NavLink } from "react-router-dom";
+(Note that we're using `parseInt` in this example - all data passed via URL
+params will be a string!)
 
-const linkStyles = {
-  display: "inline-block",
-  width: "50px",
-  padding: "12px",
-  margin: "0 6px 6px",
-  background: "blue",
-  textDecoration: "none",
-  color: "white",
-};
+In applications where your data will be contained in a `db.json` file or
+database, you'll likely want to run a `fetch` request within a `useEffect` to
+grab the specific piece of data you want from your database.
 
-function NavBar() {
-  return (
-    <div>
-      <NavLink
-        to="/"
-        exact
-        style={linkStyles}
-        activeStyle={{
-          background: "darkblue",
-        }}
-      >
-        Home
-      </NavLink>
-      <NavLink
-        to="/about"
-        exact
-        style={linkStyles}
-        activeStyle={{
-          background: "darkblue",
-        }}
-      >
-        About
-      </NavLink>
-      <NavLink
-        to="/login"
-        exact
-        style={linkStyles}
-        activeStyle={{
-          background: "darkblue",
-        }}
-      >
-        Login
-      </NavLink>
-    </div>
+Now that we have a way to access the user we want, let's updated our UserProfile
+component to display information about that user!
+
+```jsx
+// UserProfile.js
+import users from "../data.js";
+import { useParams } from "react-router-dom";
+import NavBar from "../components/NavBar";
+
+function UserProfile() {
+  const params = useParams();
+
+  const user = users.find(user => user.id === parseInt(params.id))
+
+  return(
+    <>
+      <header>
+        <NavBar />
+      </header>
+      <main>
+        <h1>{user.name}</h1>
+      </main>
+    </>
   );
-}
+};
 
-export default NavBar;
+export default UserProfile;
 ```
 
+With our component updated, we should now see the correct user displaying when
+we navigate to a specific user's profile page! Nice!
+
+>**Note**: You'll want to make sure you set up dynamic routes to work when
+>somebody shares a URL to a dynamic endpoint. There are a variety of ways to
+>handle this to make sure your app doesn't break when somebody initially loads
+>your app on a dynamic endpoint, rather than navigating to it internally. One
+>way is to use `fetch` within the component to fetch all requisite data, as
+>mentioned previously, but you might find other ways to handle it. Just remember
+>that your solution should be legible, performant, and easy to maintain!
+
```

Comment:
Yeah, perhaps this is unnecessary. Sometimes React apps aren't configured to load properly when somebody just drops a link into the browser, rather than navigate to a specific page internally. But I agree that we can remove this, as this lesson is already pretty long.

There's not specific solution or example that wouldn't require a very lengthy explanation, so I think we should just trim it.